doc/man7/provider-serializer.pod

```diff
@@ -0,0 +1,180 @@
+=pod
+
+=head1 NAME
+
+provider-serializer - The SERIALIZER library E<lt>-E<gt> provider functions
+
+=head1 SYNOPSIS
+
+ #include <openssl/core_numbers.h>
+
+ /*
+  * None of these are actual functions, but are displayed like this for
+  * the function signatures for functions that are offered as function
+  * pointers in OSSL_DISPATCH arrays.
+  */
+
+ /* Functions to serialize object data */
+ int OP_serializer_serialize_data(void *provctx, const OSSL_PARAM *data,
+                                  BIO *out);
+ int OP_serializer_serialize_object(void *provobj, BIO *out);
```

Comment:
It occurs that we might want serialization to something other than a BIO, such that the serialization is entirely internal to the provider so that (for example) private keys never leave the provider boundary. In such a case we might want some interface along the lines of "serialize this key and give me back some identifier that I can use to read it back later". Another PR though. We might need to think about the API a little bit. In some cases I imagine the "id" is provider determined (e.g. sequence number from a provider managed database), but in others the id might be client determined (e.g. key name is in some provider managed filesystem).

Reply:
I think you were thinking of deserialization (what you've called "read serialization"), right?

This has indeed occurred to me regarding how `OSSL_STORE` is supposed to works, and only for loaders that find data in serialized form.

As for a key identity, did you notice the `loadkey` function in the KEYMGMT downcalls?  It's already on my mind and will play a fairly huge part in the provider portion of `OSSL_STORE`